src/LibLog/LibLog.cs

```diff
@@ -1393,46 +1448,63 @@ internal class Log4NetLogger
-                }
-                ParameterExpression instanceParam = Expression.Parameter(typeof(object));
-                UnaryExpression instanceCast = Expression.Convert(instanceParam, loggerType);
-                ParameterExpression levelParam = Expression.Parameter(typeof(object));
-                UnaryExpression levelCast = Expression.Convert(levelParam, logEventLevelType);
-                _isEnabledForDelegate = GetIsEnabledFor(loggerType, logEventLevelType, instanceCast, levelCast, instanceParam, levelParam);
+                    // Func<object, object, bool> isEnabledFor = (logger, level) => { return ((log4net.Core.ILogger)logger).IsEnabled(level); }
+                    _loggerType = Type.GetType("log4net.Core.ILogger, log4net");
+                    if (_loggerType == null)
+                    {
+                        return;
+                    }
+                    ParameterExpression instanceParam = Expression.Parameter(typeof(object));
+                    UnaryExpression instanceCast = Expression.Convert(instanceParam, _loggerType);
+                    ParameterExpression levelParam = Expression.Parameter(typeof(object));
+                    UnaryExpression levelCast = Expression.Convert(levelParam, _logEventLevelType);
+                    _isEnabledForDelegate = GetIsEnabledFor(_loggerType, _logEventLevelType, instanceCast, levelCast, instanceParam, levelParam);
 
-                Type loggingEventType = Type.GetType("log4net.Core.LoggingEvent, log4net");
+                    Type loggingEventType = Type.GetType("log4net.Core.LoggingEvent, log4net");
 
-                _createLoggingEvent = GetCreateLoggingEvent(instanceParam, instanceCast, levelParam, levelCast, loggingEventType);
+                    _createLoggingEvent = GetCreateLoggingEvent(instanceParam, instanceCast, levelParam, levelCast, loggingEventType);
 
-                _logDelegate = GetLogDelegate(loggerType, loggingEventType, instanceCast, instanceParam);
+                    _logDelegate = GetLogDelegate(_loggerType, loggingEventType, instanceCast, instanceParam);
 
-                _loggingEventPropertySetter = GetLoggingEventPropertySetter(loggingEventType);                
+                    _loggingEventPropertySetter = GetLoggingEventPropertySetter(loggingEventType);
+                }
+                catch { }
             }
 
             [SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "ILogger")]
+            [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
             internal Log4NetLogger(dynamic logger)
             {
-                _logger = logger.Logger;
+                try
+                {
+                    _logger = logger.Logger;
+
+                    if (_logEventLevelType == null)
+                        throw new InvalidOperationException("Type log4net.Core.Level was not found.");
```

Comment:
These exceptions are just getting swallowed below?

Reply:
Yes it is mostly about making people aware, that something is not working. Since throwing exceptions in the static-constructor is not recommended.